scripts/tests/chiptest/yamltest_with_chip_repl_tester.py

```diff
@@ -0,0 +1,120 @@
+
+from chip import ChipDeviceCtrl  # Needed before chip.FabricAdmin
+import chip.FabricAdmin  # Needed before chip.CertificateAuthority
+
+# isort: on
+
+import chip.CertificateAuthority
+import chip.native
+import click
+from chip.ChipStack import *
+from chip.yaml.runner import ReplTestRunner
+from matter_yamltests.definitions import ParseSource, SpecDefinitions
+from matter_yamltests.parser import TestParser
+
+_DEFAULT_CHIP_ROOT = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", ".."))
+_CLUSTER_XML_DIRECTORY_PATH = os.path.abspath(
+    os.path.join(_DEFAULT_CHIP_ROOT, "src/app/zap-templates/zcl/data-model/"))
+
+
+def _sort_with_global_attribute_first(a, b):
+    if a.endswith('global-attributes.xml'):
+        return -1
+    elif b.endswith('global-attributes.xml'):
+        return 1
+    elif a > b:
+        return 1
+    elif a == b:
+        return 0
+    elif a < b:
+        return -1
+
+
+@click.command()
+@click.option(
+    '--setup-code',
+    default=None,
+    help='setup-code')
+@click.option(
+    '--yaml-path',
+    default=None,
+    help='yaml-path')
+@click.option(
+    '--node-id',
+    default=0x12344321,
+    help='Node ID to use when commissioning device')
+def main(setup_code, yaml_path, node_id):
+    # Setting up python environment for running YAML CI tests using python parser.
+    with tempfile.NamedTemporaryFile() as chip_stack_storage:
+        chip.native.Init()
+        chip_stack = ChipStack(chip_stack_storage.name)
+        certificate_authority_manager = chip.CertificateAuthority.CertificateAuthorityManager(
+            chip_stack, chipStack.GetStorageManager())
+        certificate_authority_manager.LoadAuthoritiesFromStorage()
+
+        if len(certificate_authority_manager.activeCaList) == 0:
+            ca = certificate_authority_manager.NewCertificateAuthority()
+            ca.NewFabricAdmin(vendorId=0xFFF1, fabricId=1)
+        elif len(certificate_authority_manager.activeCaList[0].adminList) == 0:
+            certificate_authority_manager.activeCaList[0].NewFabricAdmin(vendorId=0xFFF1, fabricId=1)
+
+        ca_list = certificate_authority_manager.activeCaList
+
+        # Creating and commissioning to a single controller to match what is currently done when
+        # running.
+        dev_ctrl = ca_list[0].adminList[0].NewController()
+        dev_ctrl.CommissionWithCode(setup_code, node_id)
+
+        # Creating Cluster definition.
+        cluster_xml_filenames = glob.glob(_CLUSTER_XML_DIRECTORY_PATH + '/*/*.xml', recursive=False)
+        cluster_xml_filenames.sort(key=functools.cmp_to_key(_sort_with_global_attribute_first))
+        sources = [ParseSource(source=name) for name in cluster_xml_filenames]
+        clusters_definitions = SpecDefinitions(sources)
+
+        # Parsing YAML test and setting up chip-repl yamltests runner.
+        yaml = TestParser(yaml_path, None, clusters_definitions)
+        runner = ReplTestRunner(clusters_definitions, dev_ctrl)
```

Comment:
Adding blocking comment since https://github.com/project-chip/connectedhomeip/pull/24349 is likely to get in first and this will need to be updated here to prevent merge race that breaks CI